uaclient/files/notices.py

```diff
@@ -160,6 +160,60 @@ def remove(self, notice_details: Notice):
         )
         system.ensure_file_absent(os.path.join(directory, filename))
 
+    def _get_notice_file_names(self, directory: str) -> List[str]:
+        """Gets the list of notice file names in the given directory.
+
+        :param directory: The directory to search for notice files.
+        :returns: List of notice file names.
+        """
+        return [
+            file_name
+            for file_name in os.listdir(directory)
+            if os.path.isfile(os.path.join(directory, file_name))
+            and self._is_valid_notice_file(directory, file_name)
+        ]
+
+    def _is_valid_notice_file(self, directory: str, file_name: str) -> bool:
+        """Checks if the notice file is valid.
+
+        :param file_name: The name of the notice file.
+        :returns: True if the file is valid, False otherwise.
+        """
+        try:
+            order_id, label = file_name.split("-")
+            for n in Notice:
+                if (
+                    n.order_id == order_id
+                    and n.label == label
+                    and (
+                        (
+                            directory == defaults.NOTICES_PERMANENT_DIRECTORY
+                            and n.is_permanent
+                        )
+                        or (
+                            directory == defaults.NOTICES_TEMPORARY_DIRECTORY
+                            and not n.is_permanent
+                        )
+                    )
+                ):
+                    return True
+            return False
+        except ValueError:
+            return False
```

Comment:
I want to point out an alternative way of implementing this to consider.
```python
is_permanent_dir = directory == defaults.NOTICES_PERMANENT_DIRECTORY
valid_file_names = {
    "{}-{}".format(n.order_id, n.label)
    for n in Notice
    if n.is_permanent == is_permanent_dir
}
return file_name in valid_file_names
```

But the current method is good and works too